src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -7485,6 +7485,48 @@ bool getILIntrinsicImplementationForRuntimeHelpers(MethodDesc * ftn,
         return true;
     }
 
+    if (tk == MscorlibBinder::GetMethod(METHOD__RUNTIME_HELPERS__GET_METHOD_TABLE)->GetMemberDef())
+    {
+        mdToken tokRawData = MscorlibBinder::GetField(FIELD__RAW_DATA__DATA)->GetMemberDef();
+
+        // In the CLR, an object is laid out as follows.
+        // [ object_header || MethodTable* (64-bit pointer) || instance_data ]
+        //                    ^                                ^-- ref <theObj>.firstField points here
+        //                    `-- <theObj> reference (type O) points here
+        //
+        // So essentially what we want to do is to turn an object reference (type O) into a
+        // native int&, then dereference it to get the MethodTable*. (Essentially, an object
+        // reference is a MethodTable**.) Per ECMA-335, Sec. III.1.5, we can add
+        // (but not subtract) a & and an int32 to produce a &. So we'll get a reference to
+        // <theObj>.firstField (type &), then back up one pointer length to get a value of
+        // essentially type (MethodTable*)&. Both of these are legal GC-trackable references
+        // to <theObj>, regardless of <theObj>'s actual length.
+
+        static BYTE ilcode[] = { CEE_LDARG_0,         // stack contains [ O ] = <theObj>
+                                 CEE_LDFLDA,0,0,0,0,  // stack contains [ & ] = ref <theObj>.firstField
+#ifdef _TARGET_64BIT_
+                                 CEE_LDC_I4_8,        // stack contains [ &, int32 ]
+#else
+                                 CEE_LDC_I4_4,        // stack contains [ &, int32 ]
+#endif
+                                 CEE_NEG,             // stack contains [ &, int32 ]
```

Comment:
This depends on constant-prop optimization. It would be better to load the right constant directly using `ldc.i4.s   -4 / -8`.

Reply:
You won't need the ifdef with `ldc.i4.s`. It can just use `-TARGET_POINTER_SIZE`.